### DISCUSSION TITLE
How to drop a packet?

### DISCUSSION THREAD

**Discussion by TheTechRobo** [ickerwx/tcpproxy#41] [Q&A] [answered]
There are some packets I want to drop entirely through a module. How would I accomplish this?

**Comment by ickerwx** [comment, marked as answer]
Hi, thanks for trying out my little tool. Dropping packets (as in, TCP packets) ist not possible. It's a TCP proxy and therefore works on streams of data. The proxy does not see packets, you are reading data through file descriptors.

Dropping packets while mitm-ing a connection is quite involved, you would need to keep track of sequence numbers and modify every subsequent packet. This is beyond the scope of this tool.

**Reply by TheTechRobo** [reply]
Oh! I should have been more clear. My module (admittedly it will be quite complicated) is going to operate on the stream and parse it into bits of data. I want to "withhold" some of it. I'm not talking about the lower-level packets.

If I just return the data from the `execute` method without that data included, would that work? Can you return an empty `bytes` object?

**Reply by ickerwx** [reply]
You can absolutely `return b''`, but I didn't know what would happen with the socket, I never actually tried this. So I ran a  quick little test. It looks like `socket.send(b'')` will not send anything, at least there was no traffic when I was sniffing with tcpdump.

**Reply by TheTechRobo** [reply]
Thanks, that's very helpful!

**Comment by ickerwx** [comment]
Moved this to a discussion thread, in case I misunderstood what you want to do.
